cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -461,13 +462,29 @@ class descriptor : public detail::descriptor_base<Task> {
 
 /// @tparam Task   Tag-type that specifies the type of the problem to solve. Can
 ///                be :expr:`task::v1::classification` or :expr:`task::v1::regression`.
+namespace dt = dal::decision_tree;
+template <typename Task>
+struct decision_tree_task_map;
+
+template <>
+struct decision_tree_task_map<task::classification> {
+    using tree_task_t = dt::task::classification;
+};
+
+template <>
+struct decision_tree_task_map<task::regression> {
+    using tree_task_t = dt::task::regression;
+};
```

Comment:
`decision_tree_task_map` should be moved to `detail`

Reply:
moved